internal/controller/virtualmachinebmc/controller.go

```diff
@@ -215,6 +245,31 @@ func (r *VirtualMachineBMCReconciler) constructServiceFromVirtualMachineBMC(virt
 	return svc
 }
 
+func (r *VirtualMachineBMCReconciler) deleteVirtBMCPod(ctx context.Context, virtualMachineBMC *bmcv1.VirtualMachineBMC) error {
+	log := log.FromContext(ctx)
+	podName := fmt.Sprintf("%s-virtbmc", virtualMachineBMC.Spec.VirtualMachineRef.Name)
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: virtualMachineBMC.Namespace,
+		},
+	}
+
+	if err := r.Delete(ctx, pod); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("virtBMC Pod already absent", "pod", podName)
+			return nil
+		}
+
+		log.Error(err, "unable to delete virtBMC Pod", "pod", podName)
+		return err
+	}
+
+	log.Info("Deleted virtBMC Pod after Secret change", "pod", podName)
```

Comment:
It feels irrelevant for this log to appear in the method. Maybe there is another better place. I suppose move it to the caller?